compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -7061,4 +7072,196 @@ private STNode parseTypeCastParam() {
 
         return STNodeFactory.createTypeCastParamNode(annot, type);
     }
+
+    /**
+     * Parse table constructor expression.
+     * <p>
+     * <code>
+     * table-constructor-expr := table [key-specifier] [ [row-list] ]
+     * </code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseTableConstructorExpr() {
+        startContext(ParserRuleContext.TABLE_CONSTRUCTOR);
+        STNode tableKeyword = parseTableKeyword();
+        STNode keySpecifier = STNodeFactory.createEmptyNode();
+        return parseTableConstructorExpr(tableKeyword, keySpecifier);
+    }
+
+    private STNode parseTableConstructorExpr(STNode tableKeyword, STNode keySpecifier) {
+        return parseTableConstructorExpr(peek().kind, tableKeyword, keySpecifier);
+    }
+
+    private STNode parseTableConstructorExpr(SyntaxKind nextTokenKind, STNode tableKeyword, STNode keySpecifier) {
+        STNode openBracket;
+        STNode rowList;
+        STNode closeBracket;
+
+        // Parse optional key specifier
+        if (nextTokenKind == SyntaxKind.KEY_KEYWORD) {
+            keySpecifier = parseKeySpecifier();
+            nextTokenKind = peek().kind;
+        }
+
+        switch (nextTokenKind) {
```

Comment:
I sense an issue here :) 

Try something like: `table key(a) foo key(b) []`

Reply:
Fixed and a test cases added.